# SF-DAT-21 | Unit Project 1


### **Read & Evaluate the following problem statement:**
Determine which free-tier customers will covert to paying customers, using demographic data collected at signup (age, gender, location, and profession) and customer useage data (days since last log in, and activity score 1 = active user, 0 = inactive user) based on Hooli data from January - April 2015.

#### **1. What is the outcome?** 

Answer: Return free-tier customer indicator (yes/no)

#### **2. What are the predictors/covariates?**

Answer: age, gender, location, profession, days since last log in, activity score (active vs. inactive user)

#### **3. What timeframe is this data relevent for?**

Answer: January - April 2015

#### **4. What is the hypothesis?**

Answer: demographic and customer usage data will allow us to predict which free-tier customers will convert to paying customers 

## **Explore Admissions Dataset**

In [2]:
import os
import numpy as np
import pandas as pd

In [3]:
df = pd.read_csv(os.path.join('..', 'dataset', 'admissions.csv'))

In [4]:
df.head()

admit    gre   gpa  prestige
0      0  380.0  3.61       3.0
1      1  660.0  3.67       3.0
2      1  800.0  4.00       1.0
3      1  640.0  3.19       4.0
4      0  520.0  2.93       4.0


#### **1. Create a data dictionary**

Variable | Description | Type of Variable
---|---|---
Admit | Admitted = 1, Not admitted = 0 | Categorical
GPA | Grade Point Average | Continuous
GRE | Score | Continuous
Prestige | 1 = lowest (not prestigious) - 4 = highest (high prestige) | Categorical


#### **2. What is the outcome?**

Answer: return applicants acceptance into UCLA graduate program (y/n)

#### **3. What are the predictors/covariates?**

Answer: admittance (y/n), GRE score, GPA, prestige (1-4)

#### **4. What timeframe is this data relevent for?**

Answer: UCLA graduate school applicants 

#### **4. What is the hypothesis?**

Answer: Using previous UCLA applicant data (admittance (1=y / 0=n), GRE score, GPA, and prestige), one can predict if an applicant will be accepted into UCLA's graduate program.  

## **Problem Statement**
*Using the above information, write a well-formed problem statement.*

Using UCLA graduate application data from previous applicants, one can determine the factors that influence UCLA admission rate and predict if a student will be accepted or not based on their GRE score, GPA and level of prestige.  


## **Exploratory Analysis Plan**
Using the lab from a class as a guide, create an exploratory analysis plan.

#### **1. What are the goals of the exploratory analysis?**


To understand the data and interpret it’s meaning.  Allows for variable exploration. Also allows for verification of the quality of the data. 

#### **2a. What are the assumptions of the distribution of data?**

Applicants with higher GRE scores, higher GPAs and higher prestige ranking are more likely to be accepted into UCLA graduate program. 


**2b. How will you determine the distribution of your data?**

Using methods to extract rows of data and running calculations on them. I.e. finding all applicants who were accepted and seeing their average GRE score, GPA and prestige ranking; then seeing which had the most weight when determining if someone was accepted.


**3a. How might outliers impact your analysis?**

Outliers could bring my averages down as well as points to extrems for deviation (min vs. max)

**3b. How will you test for outliers?**

Look at min/max and standard deviation 

Look for missing values (use 'isnull')

Look for random variables

**
4a. What is collinearity?**

When predictor variables are highlighy correlated and lead to greater accuracy.  Issues can arrise however determining the individual effects of each.

**4b. How will you test for collinearity?**

Using matplotlib


### **5. What is your exploratory analysis plan?**

*Using the above information, write an exploratory analysis plan that would allow you or a colleague to reproduce your analysis 1 year from now.*

#### Look at the number of observations in the dataframe (i.e. rows) (there are 400 rows of data = 400 observations):

In [22]:
df

admit    gre   gpa  prestige
0        0  380.0  3.61       3.0
1        1  660.0  3.67       3.0
2        1  800.0  4.00       1.0
3        1  640.0  3.19       4.0
4        0  520.0  2.93       4.0
5        1  760.0  3.00       2.0
6        1  560.0  2.98       1.0
7        0  400.0  3.08       2.0
8        1  540.0  3.39       3.0
9        0  700.0  3.92       2.0
10       0  800.0  4.00       4.0
11       0  440.0  3.22       1.0
12       1  760.0  4.00       1.0
13       0  700.0  3.08       2.0
14       1  700.0  4.00       1.0
15       0  480.0  3.44       3.0
16       0  780.0  3.87       4.0
17       0  360.0  2.56       3.0
18       0  800.0  3.75       2.0
19       1  540.0  3.81       1.0
20       0  500.0  3.17       3.0
21       1  660.0  3.63       2.0
22       0  600.0  2.82       4.0
23       0  680.0  3.19       4.0
24       1  760.0  3.35       2.0
25       1  800.0  3.66       1.0
26       1  620.0  3.61       1.0
27       1  520.0  3.74       4.0
28       1  780.0  3.22       2.0
29       0  520.0  3.29       1.0
..     ...    ...   ...       ...
370      1  540.0  3.77       2.0
371      1  680.0  3.76       3.0
372      1  680.0  2.42       1.0
373      1  620.0  3.37       1.0
374      0  560.0  3.78       2.0
375      0  560.0  3.49       4.0
376      0  620.0  3.63       2.0
377      1  800.0  4.00       2.0
378      0  640.0  3.12       3.0
379      0  540.0  2.70       2.0
380      0  700.0  3.65       2.0
381      1  540.0  3.49       2.0
382      0  540.0  3.51       2.0
383      0  660.0  4.00       1.0
384      1  480.0  2.62       2.0
385      0  420.0  3.02       1.0
386      1  740.0  3.86       2.0
387      0  580.0  3.36       2.0
388      0  640.0  3.17       2.0
389      0  640.0  3.51       2.0
390      1  800.0  3.05       2.0
391      1  660.0  3.88       2.0
392      1  600.0  3.38       3.0
393      1  620.0  3.75       2.0
394      1  460.0  3.99       3.0
395      0  620.0  4.00       2.0
396      0  560.0  3.04       3.0
397      0  460.0  2.63       2.0
398      0  700.0  3.65       2.0
399      0  600.0  3.89       3.0

[400 rows x 4 columns]

#### Look at statistical data for all UCLA applicatants (accept / reject):

In [6]:
df.describe()

admit         gre        gpa    prestige
count  400.000000  398.000000  398.00000  399.000000
mean     0.317500  588.040201    3.39093    2.486216
std      0.466087  115.628513    0.38063    0.945333
min      0.000000  220.000000    2.26000    1.000000
25%      0.000000  520.000000    3.13000    2.000000
50%      0.000000  580.000000    3.39500    2.000000
75%      1.000000  660.000000    3.67000    3.000000
max      1.000000  800.000000    4.00000    4.000000

**Count the number of acceptances = 127:**

In [17]:
df[ (df.admit == 1)].count()

admit       127
gre         127
gpa         126
prestige    126
dtype: int64

**Count the number of rejections = 273:**

In [25]:
df[ (df.admit == 0)].count()

admit       273
gre         271
gpa         272
prestige    273
dtype: int64

**Find the average 'GRE' score of those accepted = 618.90 :**

In [9]:
df['gre'][ (df.admit ==1) ].mean()

618.8976377952756

**(OR) Find ALL of the statistical data for those accepted - count, mean, standard deviation, min, quartiles, max - for each variable (GRE score, GPA, prestige): **

*Obervations:*
1. Average GRE score of those accepted = 618.90 (matches above)
2. Min GRE score of those accepted = 300 | max = 800
    1. Standard deviation of GRE = 108.88 (larger range of GRE scores for those accepted)
3. Min GPA of those accpeted = 2.42 | max = 4
    2. STD of GPA = 0.37 
4. Lowest prestige rating of those accepted = 1 | max = 4*


In [12]:
df[(df.admit == 1)].describe()

admit         gre         gpa    prestige
count  127.0  127.000000  126.000000  126.000000
mean     1.0  618.897638    3.489206    2.150794
std      0.0  108.884884    0.371655    0.921455
min      1.0  300.000000    2.420000    1.000000
25%      1.0  540.000000    3.220000    1.000000
50%      1.0  620.000000    3.545000    2.000000
75%      1.0  680.000000    3.757500    3.000000
max      1.0  800.000000    4.000000    4.000000

**Find all of the statistical data for those NOT accepted - count, mean, standard deviation, min, quartiles, max - for each variable (GRE score, GPA, prestige): **

In [18]:
df[(df.admit == 0)].describe()

admit         gre         gpa    prestige
count  273.0  271.000000  272.000000  273.000000
mean     0.0  573.579336    3.345404    2.641026
std      0.0  116.052798    0.376773    0.917198
min      0.0  220.000000    2.260000    1.000000
25%      0.0  500.000000    3.080000    2.000000
50%      0.0  580.000000    3.340000    3.000000
75%      0.0  660.000000    3.610000    3.000000
max      0.0  800.000000    4.000000    4.000000

# Bonus


#### Practice writing an alternative problem statement for your dataset:

    Using UCLA graduate application data from previous applicants, one can determine the factors that influence UCLA admission rate and predict if a student will NOT be accepted based on two or more application variables- GRE score, GPA and/or level of prestige. 

#### Risks/Assumptions:

Claiming that one or more of the variables will directly impact another is risky as there are many external factors that are not considered with this dataset (e.g. socioeconomic factors). 